### PR TITLE
fixes #6589

### LINKED ISSUE #6589: formatFloat inconsistent scientific notation
There is an inconsistency in the scientific notation produced by `formatFloat`:

```nimrod
echo formatFloat(123.456, ffScientific, precision=0)
# output on Linux:    "1.234560e+02"
# output on Windows:  "1.234560e+002"
```

The reason for that is that Visual Studio Compilers pre 2015 violate the C standard, which says: 

> "...The exponent always contains at least two digits, and only as many more digits as necessary to represent the exponent. ..." C11dr §7.21.6.1 8

See also a discussion on [StackOverflow](https://stackoverflow.com/a/31332367/1804173).

### PR DESCRIPTION
**Pull request by bluenote10**
Note: The [solutions suggested on StackOverflow](https://stackoverflow.com/a/31332367/1804173) (using `_set_output_format`) are probably not a good idea, because the function is obsolete and has been removed in the latest Visual Studio Compile. Thus, the simple post-processing should be the easiest fix.

### CODE REVIEW

File: lib/pure/strutils.nim [hunk 1]
```diff
@@ -1929,6 +1929,15 @@ proc formatBiggestFloat*(f: BiggestFloat, format: FloatFormatMode = ffDefault,
       # but nothing else is possible:
       if buf[i] in {'.', ','}: result[i] = decimalsep
       else: result[i] = buf[i]
+    when defined(windows):
```
Comment (Araq) [inline lib/pure/strutils.nim]: Is that Windows specific though or ``vcc`` specific?

Comment (bluenote10) [reply lib/pure/strutils.nim]: I'm note entirely sure, but I think VCC (whoever is providing `<stdio.h>`), and apparently all versions except the latest. Checking for VCC + version probably isn't possible, right?

But if the implementation is already correct, the cheap check `result[^4] == '+'` will simply fail, which is negligible in terms of performance of the whole function.

Comment (Yardanico) [reply lib/pure/strutils.nim]: Maybe check for VCC at least?
when defined(vcc)

Comment (bluenote10) [reply lib/pure/strutils.nim]: Thanks, I missed that there is `defined(vcc)`, was looking for `vcs` ;). Fixed now.

**Pull request merged**